app/Model/ActivityFilter.php

```diff
@@ -0,0 +1,400 @@
+     */
+    public function filterByCreationDate($date)
+    {
+        if ($date === 'recently') {
+            return $this->filterRecentlyDate(ProjectActivity::TABLE.'.date_creation');
+        }
+
+        return $this->filterWithOperator(ProjectActivity::TABLE.'.date_creation', $date, true);
+    }
+
+    /**
+     * Filter by creation date
+     *
+     * @access public
+     * @param  string  $start
+     * @param  string  $end
+     * @return ActivityFilter
+     */
+    public function filterByCreationDateRange($start, $end)
+    {
+        $this->query->addCondition($this->getCalendarCondition(
+            $this->dateParser->getTimestampFromIsoFormat($start),
+            $this->dateParser->getTimestampFromIsoFormat($end),
+            'date_creation',
+            'date_completed'
+        ));
+
+        return $this;
+    }
+
+    /**
+     * Filter by comments
+     *
+     * @access public
+     * @param  string  $string
+     * @return ActivityFilter
+     */
+    public function filterByComment($string) {
+        $this->query->beginOr();
+
+        $this->query->ilike(Comment::TABLE.'.comment', '%'.$string.'%');
+        $this->query->join(Comment::TABLE, 'task_id', 'task_id');
+
+        $this->query->closeOr();
+    }
+
+    /**
+     * Get all results of the filter
+     *
+     * @access public
+     * @return array
+     */
+    public function findAll()
+    {
+        return $this->query->asc(Task::TABLE.'.id')->findAll();
+    }
+
+    /**
+     * Get the PicoDb query
+     *
+     * @access public
+     * @return \PicoDb\Table
+     */
+    public function getQuery()
+    {
+        return $this->query;
+    }
+
+    /**
+     * Filter with an operator
+     *
+     * @access public
+     * @param  string    $field
+     * @param  string    $value
+     * @param  boolean   $is_date
+     * @return ActivityFilter
+     */
+    private function filterWithOperator($field, $value, $is_date)
+    {
+        $operators = array(
+            '<=' => 'lte',
+            '>=' => 'gte',
+            '<' => 'lt',
+            '>' => 'gt',
+        );
+
+        foreach ($operators as $operator => $method) {
+            if (strpos($value, $operator) === 0) {
+                $value = substr($value, strlen($operator));
+                $this->query->$method($field, $is_date ? $this->dateParser->getTimestampFromIsoFormat($value) : $value);
+                return $this;
+            }
+        }
+
+        if ($is_date) {
+            $timestamp = $this->dateParser->getTimestampFromIsoFormat($value);
+            $this->query->gte($field, $timestamp);
+            $this->query->lte($field, $timestamp + 86399);
+        } else {
+            $this->query->eq($field, $value);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get the event html content
+     *
+     * @access public
+     * @param  array     $params    Event properties
+     * @return string
+     */
+    public function getContent(array $params)
```

Comment:
This method and the one below are duplicated from the class `ProjectActivity`, there is probably a better way of doing that.